packages/contracts/src/manager/libraries/LibDiamond.sol

```diff
@@ -0,0 +1,238 @@
+        mapping(bytes4 => FacetAddressAndPosition) selectorToFacetAndPosition;
+        // maps facet addresses to function selectors
+        mapping(address => FacetFunctionSelectors) facetFunctionSelectors;
+        // facet addresses
+        address[] facetAddresses;
+        // Used to query if a contract implements an interface.
+        // Used to implement ERC-165.
+        mapping(bytes4 => bool) supportedInterfaces;
+        // owner of the contract
+        address contractOwner;
+    }
+
+    function diamondStorage() internal pure returns (DiamondStorage storage ds) {
+        bytes32 position = DIAMOND_STORAGE_POSITION;
+        assembly {
+            ds.slot := position
+        }
+    }
+
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+    function setContractOwner(address _newOwner) internal {
+        DiamondStorage storage ds = diamondStorage();
+        address previousOwner = ds.contractOwner;
+        ds.contractOwner = _newOwner;
+        emit OwnershipTransferred(previousOwner, _newOwner);
+    }
+
+    function contractOwner() internal view returns (address contractOwner_) {
+        contractOwner_ = diamondStorage().contractOwner;
+    }
+
+    function enforceIsContractOwner() internal view {
+        require(msg.sender == diamondStorage().contractOwner, "LibDiamond: Must be contract owner");
+    }
+
+    event DiamondCut(IDiamondCut.FacetCut[] _diamondCut, address _init, bytes _calldata);
+
+    function addDiamondFunctions(
+        address _diamondCutFacet,
+        address _diamondLoupeFacet,
+        address _ownershipFacet
+    ) internal {
+        IDiamondCut.FacetCut[] memory cut = new IDiamondCut.FacetCut[](3);
+        bytes4[] memory functionSelectors = new bytes4[](1);
+        functionSelectors[0] = IDiamondCut.diamondCut.selector;
+        cut[0] = IDiamondCut.FacetCut({facetAddress: _diamondCutFacet, action: IDiamondCut.FacetCutAction.Add, functionSelectors: functionSelectors});
+        functionSelectors = new bytes4[](5);
+        functionSelectors[0] = IDiamondLoupe.facets.selector;
+        functionSelectors[1] = IDiamondLoupe.facetFunctionSelectors.selector;
+        functionSelectors[2] = IDiamondLoupe.facetAddresses.selector;
+        functionSelectors[3] = IDiamondLoupe.facetAddress.selector;
+        functionSelectors[4] = IERC165.supportsInterface.selector;
+        cut[1] = IDiamondCut.FacetCut({
+            facetAddress: _diamondLoupeFacet,
+            action: IDiamondCut.FacetCutAction.Add,
+            functionSelectors: functionSelectors
+        });
+        functionSelectors = new bytes4[](2);
+        functionSelectors[0] = IERC173.transferOwnership.selector;
+        functionSelectors[1] = IERC173.owner.selector;
+        cut[2] = IDiamondCut.FacetCut({facetAddress: _ownershipFacet, action: IDiamondCut.FacetCutAction.Add, functionSelectors: functionSelectors});
+        diamondCut(cut, address(0), "");
```

Comment:
consider a replacement by
```
    DiamondInit dInit = new DiamondInit();
        diamondCut(
            cut,
            address(dInit),
            abi.encodeWithSelector(DiamondInit.init.selector)
        );
        ```

Reply:
Yeah, feeling we should make a good use of init() function